doc/sfp-cmis/cmis-init.md

```diff
@@ -0,0 +1,447 @@
+
+## Functional Requirements
+
+1. Ability to parse the advertised applications from the transceivers
+2. Ability to post the advertised applications to the STATE_DB
+3. Ability to support multiple CMIS transceivers in one single thread.
+4. Ability to detect the errors of the application initialization on the transceivers
+
+**Note:**  
+The duration of the CMIS application initialization greatly differs from transceivers
+to transceivers, while some take 3 seconds for activating the 4x100G mode, some take 15 seconds.
+
+## Warm Boot Requirements
+
+Functionality should continue to work across warm boot. 
+- The CMIS application initialization should not be performed during WarmBoot.
+- The CMIS application initialization should be skipped if no application code updates.
+
+# Functional Description
+
+- The **pmon#xcvrd** should detect the module type of the attched CMIS transceiver and
+post its module information onto the **STATE_DB**
+- When a CMIS transceiver is attached, the **show interfaces transceiver eeprom**
+should display the advertised applications of the transceiver.
+- When a CMIS transceiver is attached, the **sfputil** should be capable of reporting
+the errors of the CMIS application initialization.
+- When a CMIS transceiver is detected, the **pmon#xcvrd** should automatically activate
+the appropriate application as per the current port mode.
+- Only the default application mode will be supported for now, we'll deal with the dynamic
+application update later when the synchronization mechanism between **syncd** and **pmon#xcvrd**
+is in place and ready.
+
+# Design
+
+## sonic-platform-daemons/sonic-xcvrd (modified)
+
+The transceiver daemon will be enhanced as below:
+
+- The **application_advertisement** in the **TRANSCEIVER_INFO** table is now updated
+to the json format:  
+
+**Original:**  
+```
+  "TRANSCEIVER_INFO|Ethernet0": {
+    "type": "hash",
+    "value": {
+      "application_advertisement": "400GAUI-8 C2M (Annex 120E) - 400GBASE-DR4 (Cl 124)\n        100GAUI-2 C2M (Annex 135G) - 100G-FR/100GBASE-FR1 (Cl 140)",
+      ...... omitted ......
+    }
+  },
+```
+
+**Modified:**  
+```
+  "TRANSCEIVER_INFO|Ethernet0": {
+    "type": "hash",
+    "value": {
+      "application_advertisement": "{1: {'host_electrical_interface_id': '400GAUI-8 C2M (Annex 120E)', 'module_media_interface_id': '400GBASE-DR4 (Cl 124)', 'host_lane_count': 8, 'media_lane_count': 4, 'host_lane_assignment_options': 1, 'media_lane_assignment_options': None}, 2: {'host_electrical_interface_id': '100GAUI-2 C2M (Annex 135G)', 'module_media_interface_id': '100G-FR/100GBASE-FR1 (Cl 140)', 'host_lane_count': 2, 'media_lane_count': 1, 'host_lane_assignment_options': 85, 'media_lane_assignment_options': None}}",
+      ...... omitted ......
+    }
+  },
+  ...... omitted ......
+```  
+
+- **xcvrd_utilities/port_mapping.py**: Add support for **APPL_DB** and **STATE_DB**
```

Comment:
Please add a section for "Conditions for Datapath init". Please specify in which all scenarios the datapath will be initialized or re-initialized

Reply:
; how will the CLI look like? sample output?
It's available in the following code PR
https://github.com/Azure/sonic-utilities/pull/1855

; Please add a section for "Conditions for Datapath init" ....
Sure, will do